src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java

```diff
@@ -515,8 +518,15 @@ public Submission findLatestSubmissionWithRatedResultWithCompletionDate(Particip
         if (participation.getSubmissions() == null || participation.getSubmissions().isEmpty()) {
             return null;
         }
+
+        Result result = null;
+        List<Result> results;
+        List<Result> latestResults;
         for (var submission : participation.getSubmissions()) {
-            var result = submission.getResult();
+            results = submission.getResults();
```

Comment:
why do you duplicate the same code all the time? This is really really bad! As an object oriented programmer, you should think first about ___`reuse`___
It is really really important that we do not simply copy and paste code around. Please fix this before you continue with your PR, otherwise I will close it!